B/Botan/build_tarballs.jl

```diff
@@ -0,0 +1,120 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "Botan"
+version = v"3.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource(
+        "https://botan.randombit.net/releases/Botan-$(version).tar.xz",
+        "71843afcc0a2c585f8f33fa304f0b58ae4b9c5d8306f894667b3746044277557",
+    ),
+    FileSource(
+        "https://github.com/joseluisq/macosx-sdks/releases/download/13.3/MacOSX13.3.sdk.tar.xz",
+        "518e35eae6039b3f64e8025f4525c1c43786cc5cf39459d609852faf091e34be",
+    ),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+target_triplet=(${target//-/ })
+
+# The preinstalled MacOSX SDK lacks support for C++20.
+# Replace it with a newer version that has full support.
+if [[ "${target}" == *"apple-darwin"* ]]; then
+    cd ${WORKSPACE}/srcdir
+    tar xvf MacOSX*.sdk.tar.xz
+    mount --bind MacOSX*.sdk /opt/${target}/${target}/sys-root
+    export MACOSX_DEPLOYMENT_TARGET=$(echo MacOSX*.sdk | grep -oP '\d+\.\d')
+fi
+
+# BinaryBuilder doesn't allow using "-march" even where it makes sense.
+# Remove the code from the ${CXX} wrapper that is responsible for this.
+if [[ "${target}" == *"aarch64"* ]]; then
+    sed -i '/Cannot force an architecture via -march/,+1 s/^/: #/' $(which ${CXX})
+fi
+
```

Comment:
This was my first workaround. But then `-march` will only be set for two files that need specific `SHA3` and `SM4` instructions (whose availability will be checked at runtime).

I assumed that there is a good reason that `BinaryBuilder` has `lock_microarchitectures = true` by default? This is why I came up with the solution above where `-march` will be set everywhere.

Reply:
Note that I only disabled the error, i.e. the workaround is not the same as setting `lock_microarchitectures = false`. `BinaryBuillder` will still set `-march` everywhere. Compiling for `aarch64` this will cause two compiler warnings (which I intentionally didn't remove), where the second `-march` set by `Botan` takes precedence.

Reply:
> I assumed that there is a good reason that BinaryBuilder has lock_microarchitecture = true by default? 

Of course it does: it's to prevent build systems from using `-march=...` which in most cases generates non generic binaries, which would result in `SIGILL` for users at runtime.

Reply:
Sorry, I should've expressed myself more clearly.

As far as I can see, `lock_microarchitecture = true` does two things:

1. Setting `-march`.
2. Preventing the build system from using `-march`.

So what I wanted to say is:

> I assumed that there is a good reason that `BinaryBuilder` has `lock_microarchitecture = true` **so that 1. is carried out** by default?

To summarize, I used the proposed workaround instead of setting `lock_microarchitecture = false` because I wanted to disable 2. without disabling 1. at the same time for no good reason.